controllers/capabilities/proxy.go

```diff
@@ -200,6 +200,10 @@ func (t *ProductThreescaleReconciler) syncProxyOIDC(params threescaleapi.Params,
 	// If plain value is not nil - use plain value as precedence over secret
 	issuerEndpoint := oidcSpec.IssuerEndpoint
 	if issuerEndpoint == "" {
+		if oidcSpec.IssuerEndpointRef == nil {
+			// If missing both IssuerEndpoint and  IssuerEndpointRef in OpenApi CR - Product will fail SyncProxy
+			return fmt.Errorf("missing IssuerEndpoint definition in OIDC spec in openapi CR. Product OpenID Connect Issuer will not be set.")
```

Comment:
Hey @valerymo - this should be a validationError IMO and the reconciler should not be re-triggered, instead, it should wait for user change to the CR and re-reconcile then

Reply:
hey @MStokluska . This Task was based on 10518, where the fix was done. Now after https://github.com/3scale/3scale-operator/pull/915 PR (task 10518) merge  - issue resolved and proxy.go file disapered from the list of changes in this PR. Tasks 10523 and 10524 are dependent to 10518 and mostly resolved there (in 10518). The main goal of current task - make the process more clear for user (documentation) Thank you for comment

Reply:
@valerymo I see, thanks. But please update the error handling as part of this PR since it got pretty bad with the reconciliation, alternatively open another PR. We are about to release so we either need the error fixed or the previous PR reverted.